### PR TITLE
[2015.2] Alias `cmd.run` to `cmd.shell` in templates (python_shell=True)

### PR DESCRIPTION
**Pull request by basepi**
Includes some new functionality to FunctionWrapper in salt-ssh to make this possible.

### CODE REVIEW

**Comment by thatch45** [comment]
https://jenkins.saltstack.com/job/salt-pr-rs-cent7-n/3128/testReport/junit/integration.cli.custom_module/SSHCustomModuleTest/test_ssh_sls_with_custom_module/

Could that be related?

**Comment by basepi** [comment]
Unsure. I will investigate.

**Comment by basepi** [comment]
Go Go Jenkins!

**Comment by basepi** [comment]
I don't think this is related. It's custom modules, and this is what the SLS looks like:

```
regular-module:
  module.run:
    - name: test.echo
    - text: hello

custom-module:
  module.run:
    - name: test.recho
    - text: hello
```

No jinja, so I doubt it's related, sounds like it's probably a fluke (ssh couldn't connect or something, so we got an error back instead of the state dictionary). Time will tell, I restarted jenkins.

**Comment by basepi** [comment]
Huh, same failures. Guess it could be related. I'll try to reproduce.

**Comment by basepi** [comment]
Yep, it's a valid error. FunctionWrapper doesn't support assignment.

```
test:
    - Rendering SLS custom_module failed, render error: 'FunctionWrapper' object does not support item assignment
    - Traceback (most recent call last):
    -   File "/usr/local/lib/python2.7/dist-packages/salt/state.py", line 2504, in render_state
    -     sls, rendered_sls=mods
    -   File "/usr/local/lib/python2.7/dist-packages/salt/template.py", line 79, in compile_template
    -     ret = render(input_data, saltenv, sls, **render_kwargs)
    -   File "/usr/local/lib/python2.7/dist-packages/salt/renderers/jinja.py", line 327, in render
    -     **kws)
    -   File "/usr/local/lib/python2.7/dist-packages/salt/utils/templates.py", line 61, in render_tmpl
    -     kws['salt']['cmd.run'] = kws['salt']['cmd.shell']
    - TypeError: 'FunctionWrapper' object does not support item assignment
```

**Comment by basepi** [comment]
Got it! This is working in salt-ssh now. =)

File: salt/utils/templates.py [hunk 1]
```diff
@@ -54,6 +54,15 @@ def render_tmpl(tmplsrc,
...
             context = {}
 
+        # Alias cmd.run to cmd.shell to make python_shell=True the default for
+        # templated calls
+        if 'salt' in kws:
+            if 'cmd.run' in kws['salt'] and 'cmd.shell' in kws['salt']:
+                kws['salt']['cmd.run'] = kws['salt']['cmd.shell']
+            if 'run' in kws['salt'].get('cmd', {}) \
+                    and 'shell' in kws['salt'].get('cmd', {}):
+                kws['salt']['cmd']['run'] = kws['salt']['cmd']['shell']
```
Comment (cachedout) [inline salt/utils/templates.py]: Will this fail if the cmd key isn't present?

Comment (basepi) [inline salt/utils/templates.py]: No. Note in the previous line, I do `if 'run' in kws['salt'].get('cmd', {})`. If `cmd` isn't present, then the `.get()` will return an empty dict, and the `if 'run' in` will fail.

**Comment by basepi** [comment]
Go Go Jenkins!

**Comment by basepi** [comment]
There's something going on here with salt-ssh, I'm going to investigate more tomorrow.

**Comment by rallytime** [comment]
Go Go Jenkins!

**Comment by basepi** [comment]
I think that ubuntu failure is real. Investigating.

**Comment by basepi** [comment]
That should do it, go tests go!

**Pull request merged**